### ISSUE TITLE
github-actions bot not closing issues.

### ISSUE THREAD

**Issue by ditman** [flutter/flutter#158356]
### Type of Request

bug

### Infrastructure Environment

github-actions bot

### What is happening?

I marked an issue as "waiting for customer response" 3 weeks ago, and after a period of inaction, the `github-actions` bot messaged that it was going to "reluctantly going to close this bug for now".

It didn't actually close the issue though (I guess it was feeling *very* reluctant!)

Here's the link to the issue:

* https://github.com/flutter/flutter/issues/155444

I'm leaving it open in case there's something fishy on it.

### Steps to reproduce

1. Add the `waiting for customer response` label to a flutter/flutter issue.
2. Wait 3 weeks.

### Expected results

I'd expect the issue to be *closed* after the inactivity period. The issue where I applied this label remained open.

**Comment by christopherfujino** [comment]
No idea why this didn't work, but as we're using a forked action: https://github.com/flutter/flutter/blob/master/.github/workflows/no-response.yaml

We should migrate this to using the upstream one.

**Comment by ditman** [comment]
This issue looks like is going to go unattended for 3 weeks:

* https://github.com/flutter/flutter/issues/157014

**Comment by ditman** [comment]
I looked at the latest action logs:

* https://github.com/flutter/flutter/actions/runs/11733733863/job/32688540167

And it seems that it's choking at a locked issue:

```
flutter/flutter#149165 is being closed
/home/runner/work/_actions/godofredoc/no-response/0ce2dc0e63e1c7d2b87752ceed091f6d32c9df09/dist/index.js:3738
      const error = new requestError.RequestError(toErrorMessage(data), status, {
                    ^

RequestError [HttpError]: Unable to create comment because issue is locked.
    at /home/runner/work/_actions/godofredoc/no-response/0ce2dc0e63e1c7d2b87752ceed091f6d32c9df09/dist/index.js:3738:21
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  status: 403,
```

Issue:

* https://github.com/flutter/flutter/issues/149165

---

I'm going to try and remove the label from that issue and re-run the action, and see if it can continue. (I also wonder why the action attempted to run in a closed issue)

**Comment by stuartmorgan** [comment]
Thanks for investigating this! Primary triage brought this up recently but I hadn't been able to look into it yet. I forgot we have run logs for this that I could look at.

**Comment by stuartmorgan** [comment]
> We should migrate this to using the upstream one.

We were not able to get fixed landed upstream that we needed for it to work for us (IIRC it wasn't correctly handling repos with lots of issues); unless that changed we'll need to re-fork instead.

**Comment by stuartmorgan** [comment]
Yes, here we go: https://github.com/lee-dohm/no-response/issues/211

We got the fix landed, but couldn't use it without more changes that we couldn't get landed. Note that the last commit to the upstream repo is 3 years ago.

**Comment by ditman** [comment]
Link to all failed jobs:

* https://github.com/flutter/flutter/actions/workflows/no-response.yaml?query=is%3Afailure

(There's a lottttt of them that link back to the same locked issue, probably the code of the action needs to be updated to handle that case? Or maybe be more clear when failing?)

**Comment by christopherfujino** [comment]
> Yes, here we go: [lee-dohm/no-response#211](https://github.com/lee-dohm/no-response/issues/211)
> 
> We got the fix landed, but couldn't use it without more changes that we couldn't get landed. Note that the last commit to the upstream repo is 3 years ago.

Thanks for the context.

**Comment by ditman** [comment]
The bot seems to be closing issues again (?)

* https://github.com/flutter/flutter/issues/155406#issuecomment-2463849291
